extensions/amp-analytics/0.1/visibility-manager.js

```diff
@@ -376,9 +378,27 @@ export class VisibilityManager {
           'elementHeight': layoutBox.height,
         }));
       }
+
+      // If spec includes initial scroll depth, set it
+      if (spec['initialScrollDepth']) {
+        model.maybeSetInitialScrollDepth(viewport.getScrollTop());
```

Comment:
Is it too late to setInitialScrollDepth at trigger time? I believe you want to set the initalScrollDepth at element ini-load time, that will be at the `readyPromise` resolve time.

Reply:
Good idea - done.